tests/brokermetadata/BrokerMetadataTest.cc

```diff
@@ -25,19 +25,58 @@ using namespace pulsar;
 TEST(BrokerMetadataTest, testConsumeSuccess) {
     Client client{"pulsar://localhost:6650"};
     Producer producer;
-    Result producerResult = client.createProducer("persistent://public/default/testConsumeSuccess", producer);
+    ProducerConfiguration producerConfiguration;
+    producerConfiguration.setBatchingEnabled(false);
```

Comment:
The only difference between these two tests is here. I think you can use `INSTANTIATE_TEST_SUITE_P` to reduce duplicated code. See `AcknowledgeTest.cc` for example.

But it's only an improvement. If you think it's too complicated, just resolve this comment.

Reply:
@BewareMyPower I am fine with this improvement. Since we need to use async, I think we don't need this improvement